Add bash script to check driver on some distros.

aquadzn opened the pull request.

aquadzn committed:
:pencil: Add bash script to check and install needed driver

aquadzn committed:
:heavy_plus_sign: Add instruction for script in README

NicolasGuilloux commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,127 @@
+            echo -e "\t* Use the Arekinath patch to provides VA API compatibility for recent NVIDIA cards."
+        fi
+
+        if $(vainfo |& grep -q -e "H265" -e "HEVC")
```
It would be great to all of the checks inside a function to avoid boiler plate

aquadzn committed:
Fix Fedora case and wrap some redundant lines into a function

aquadzn commented:
done 👍

agentcobra reviewed (approved):
look OK for me

aquadzn committed:
:pencil: Also wrap redundant Nvidia lines

NicolasGuilloux commented:
There is some stuff that I dislike in the script. It tries to install the driver evenif the vainfo which is very instrusive if we widespread this script for all community.

A better approach would be to test vainfo, and if it is not optimal, it checks if all the appropriate drivers are installed. If not, it asks to actually install them and recheck.

Moreover, a small warning about why we use `sudo` would be great aswell. Many users of the scripts we made are not good Linux users, and I guess they will start the command several times and enter their password several times aswell. Limiting that would be great :)

NicolasGuilloux commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,98 @@
+    "Ubuntu" | "Debian GNU/Linux")
+        if ! type "vainfo" > /dev/null
+        then
+            sudo apt-get update && sudo apt-get install vainfo libva-glx2
```
`sudo` warning (cf PR comment)

NicolasGuilloux commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,98 @@
+
+        check_codec
+        ;;
+esac
```
No fallback for unrecognized OS ?

NicolasGuilloux commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,98 @@
+nvidia() {
+    echo -e "For NVIDIA users, you have 2 options:\n"
+    echo -e "\t* Use the Nouveau driver and install the old NVIDIA firmware that provides support for the VA API."
+    echo -e "\t* Use the Arekinath patch to provides VA API compatibility for recent NVIDIA cards."
```
A link to the Arekinath patch

NicolasGuilloux commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,98 @@
+    then
+        echo -e "- H264 found."
+    else
+        echo -e "\nH264 not found.\nPlease generate a report with the following command:\n\t- curl https://raw.githubusercontent.com/NicolasGuilloux/blade-shadow-beta/master/scripts/report.pl | perl"
```
`Sadly, not supporting at least H264 means not supporting Shadow at all. To get some help, please generate a full report with the following command, and share it with the Discord community`

NicolasGuilloux commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,98 @@
+
+
+check_codec() {
+    if $(vainfo |& grep -q -e "H265" -e "HEVC")
```
I don't know if it is a sufficient condition to have H265|HEVC inside the vainfo to actually support the H265 codec. These are 2 regex I used to check H265 and H264:

- `/VAProfileH264Main.*:\t(VAEntrypoint\w*)/`
- `/VAProfile(HEVC|H265)Main.*:\t(VAEntrypoint\w*)/`

NicolasGuilloux commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,98 @@
+    "Fedora")
+        if ! type "vainfo" > /dev/null
+        then
+            sudo dnf install librtmp libva-intel-hybrid-driver libvdpau-va-gl
```
Does it work for AMD Radeon ?

aquadzn replied:
Can't find any infos about Radeon so I skipped it

aquadzn committed:
Refactor all script.

aquadzn commented on scripts/check_driver.sh:
````diff
@@ -0,0 +1,230 @@
+# ```bash check_driver.sh```
+
+
+check_h265_hevc() {
+    if $(vainfo |& grep -q -E "VAProfile((H265|HEVC))")
+    then
+        true
+    else
+        false
+    fi
+}
+
+check_h264() {
+    if $(vainfo |& grep -q -E "VAProfileH264Main")
+    then
+        true
+    else
+        echo -e "❌ \e[91mH264 not found.\e[0m\nPlease generate a report with the following command:\n\t- curl https://raw.githubusercontent.com/NicolasGuilloux/blade-shadow-beta/master/scripts/report.pl | perl"
+        false
+    fi
+}
````
I changed the grep pattern and wrapped both into two functions

aquadzn commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,230 @@
+}
+
+
+print_nvidia() {
+    echo -e "For NVIDIA users, you have 2 options:\n"
+    echo -e "\t* Use the Nouveau driver and install the old NVIDIA firmware that provides support for the VA API."
+    echo -e "\t* Use the Arekinath patch to provides VA API compatibility for recent NVIDIA cards."
+    echo -e "\t* https://gitlab.com/aar642/libva-vdpau-driver"
+}
```
Added URL to Nvidia warning

aquadzn commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,230 @@
+}
+
+
+print_distro() {
+    echo -e "✅ \e[32mFound $1.\e[0m"
+}
```
Added function to print the distro

aquadzn commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,230 @@
+}
+
+
+print_vainfo_warning() {
+    echo -e "❌ \e[91mvainfo not found.\e[0m\n"
+    echo -e "⏸️  \e[93mNeed permission to install vainfo. Please re-run the script after installation.\e[0m\n"
+    sleep 2
+}
+
+print_driver_warning() {
+    echo -e "⏸️  \e[93mNeed permission to install driver(s).\e[0m\n"
+    sleep 2
+}
```
Added two warnings for vainfo and drivers installation

aquadzn commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,230 @@
+
+
+main() {
+    echo -e "\n⚠️ \e[93mWarning\033[0m\nThis script will ask for sudo permission in order to install missings packages and/or drivers.\n"
+    sleep 3
```
Added a warning about the use of sudo

aquadzn commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,230 @@
+
+            fi
+            ;;
+        *)
+            echo -e "❌ \e[91mDistribution not found. Please try again.\e[0m\n"
+            return
+            ;;
```
Added a case when distro is not in case choices

aquadzn reviewed (commented):
Good | Missing vainfo | Distro not found
--- | --- | ---
![good](https://user-images.githubusercontent.com/46140458/80734298-3fd64a00-8b0f-11ea-86e6-307818bd38fc.png) | ![vainfo](https://user-images.githubusercontent.com/46140458/80734461-7f049b00-8b0f-11ea-87d3-cf36c6370786.png) | ![distro](https://user-images.githubusercontent.com/46140458/80734556-a0fe1d80-8b0f-11ea-8772-10dc210e8466.png)

@NicolasGuilloux @agentcobra 😃

aquadzn committed:
export LIBVA_DRIVER_NAME=i965 for Fedora distro

aquadzn commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,230 @@
+                if [[ $pilot == *"Intel"* ]]
+                then
+                    sudo dnf install libva-intel-hybrid-driver libvdpau-va-gl
+                    sudo sh -c "echo 'export LIBVA_DRIVER_NAME=i965' >> /etc/profile"
```
#55

NicolasGuilloux replied:
I disagree: all Intel GPU does not support the `i965`, and the `iHD` driver may be better as it should give better performance.

aquadzn replied:
reverted https://github.com/aquadzn/blade-shadow-beta/commit/945c0f8bf0ef034dcdf4df102163e4f42819b931

PhilippeMA committed:
Fedora test for vainfo

There was a mistake on the test.

aquadzn committed:
Merge pull request #1 from PhilippeMA/patch-1

Fedora test for vainfo

aquadzn committed:
spell mistake: vaino >>> vainfo

NicolasGuilloux commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,230 @@
+                fi
+                if [[ ! check_h264 ]]
+                then
+                    echo -e "❌ \e[91mH264 not found.\e[0m\nPlease generate a report with the following command:\n\t- curl https://raw.githubusercontent.com/NicolasGuilloux/blade-shadow-beta/master/scripts/report.pl | perl"
```
You already print this warning from the `check_h264` function, don't you? :p

aquadzn replied:
https://github.com/NicolasGuilloux/blade-shadow-beta/pull/54#discussion_r418288125

NicolasGuilloux commented on scripts/check_driver.sh:
```diff
@@ -0,0 +1,230 @@
+                fi
+                if [[ ! check_h264 ]]
+                then
+                    echo -e "❌ \e[91mH264 not found.\e[0m\nPlease generate a report with the following command:\n\t- curl https://raw.githubusercontent.com/NicolasGuilloux/blade-shadow-beta/master/scripts/report.pl | perl"
```
Same here, you print it from the function in the if

aquadzn committed:
Revert "export LIBVA_DRIVER_NAME=i965 for Fedora distro"

This reverts commit ebf448aef293bc971c0962ea9b2d461186817b97.

aquadzn committed:
Removed the echo line from check_h264()

aquadzn commented on scripts/check_driver.sh:
```diff
@@ -17,7 +17,6 @@ check_h264() {
     then
         true
     else
-        echo -e "❌ \e[91mH264 not found.\e[0m\nPlease generate a report with the following command:\n\t- curl https://raw.githubusercontent.com/NicolasGuilloux/blade-shadow-beta/master/scripts/report.pl | perl"
```
I forgot to remove it from the function (it wasn't printing)

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 1,216.68kb -> 1,009.54kb (17.03%)

/docs/assets/css/images/overlay.png -- 19.63kb -> 2.76kb (85.95%)
/docs/images/logo.png -- 27.42kb -> 17.03kb (37.89%)
/docs/images/share_folder.png -- 354.91kb -> 248.44kb (30%)
/docs/images/appimage.png -- 22.24kb -> 17.33kb (22.08%)
/docs/images/hamachi_virtualhere.png -- 118.13kb -> 93.61kb (20.76%)
/docs/images/dolphin_share.png -- 53.87kb -> 44.56kb (17.29%)
/docs/images/hamachi_config.png -- 90.20kb -> 76.13kb (15.59%)
/docs/images/hamachi_virtualhere_linux.png -- 54.47kb -> 46.25kb (15.09%)
/docs/images/shadow-tool.png -- 24.08kb -> 20.71kb (14%)
/docs/assets/css/styles/brown-papersq.png -- 17.77kb -> 17.36kb (2.32%)
/docs/assets/fonts/fontawesome-webfont.svg -- 433.96kb -> 425.36kb (1.98%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

aquadzn committed:
Merge pull request #2 from aquadzn/imgbot

Reduce images total size by 17%
